### PR TITLE
fixed the failing user tests - negative create, test_user_list

### PR DESCRIPTION
**Pull request by tkolhar**
please review it
thanks

### CODE REVIEW

File: tests/foreman/cli/test_user.py [hunk 1]
```diff
@@ -646,7 +646,6 @@ def test_negative_create_user_3(self, opts):
         'email@brazil.b',
         '{0}@example.com'.format(
             gen_string('alpha', 49)),  # total length 61
-        '',
```
Comment (elyezer) [inline tests/foreman/cli/test_user.py]: Empty email is now being accepted?

File: tests/foreman/cli/test_user.py [hunk 2]
```diff
@@ -675,6 +674,36 @@ def test_negative_create_user_4(self, email):
...
+    @data(
+        ''
+    )
+    def test_bugzilla_1204686(self, email):
+        """@Test: Create User with invalid Email Address
+
+        @Feature: User - Negative Create
+
+        @Steps:
+        1. Create User for all invalid Email Address in [2]
```
Comment (elyezer) [inline tests/foreman/cli/test_user.py]: This test is just using one data entry, then the `@data` decorator is not needed you can use the `''` string in the test body. Like:

```
options = {
    # ...
    'mail': '',
    # ...
}

**Comment by elyezer** [comment]
I can't get the idea of this PR, will NACK until I get a better understanding of what is the idea. By lokking here I can assume that for any reason passing the `''` as the email address is not working but nothing more. Also the test which checks that failure needs to get its docstring updated.

File: tests/foreman/cli/test_user.py [hunk 3]
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
         self.assertNotEqual(result.return_code, 0)
         self.assertTrue(result.stderr)
 
+    @skip_if_bug_open('bugzilla', 1204686)
+    def test_bugzilla_1204686(self, email):
```
Comment (elyezer) [inline tests/foreman/cli/test_user.py]: This test will fail because was left the `email` argument.

**Comment by tkolhar** [comment]
@elyezer : empty  email addresses should not be accepted but they are working hence
i filed the bz : 1204686 and added the test

Second scenario: email addresses with latin and utf-8 should be accepted but they are not 
working hence i filed the bz : 1204667 and added the test

File: tests/foreman/cli/test_user.py [hunk 4]
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
...
+    def test_bugzilla_1204686(self):
+        """@Test: Create User with Empty Email Address
+
+        @Feature: User - Negative Create
+
+        @Steps:
+        1. Create User with empty Email Address in [2]
+        using valid Username, First Name, Surname, Language, authorized by
+
+        @Assert: User is not created. Appropriate error shown.
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_user.py]: How are you verifying that an appropriate error is shown? You can use `self.assertIn('desired error message', result.stdout)` to verify that some particular error message is shown.

File: tests/foreman/cli/test_user.py [hunk 5]
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
...
+            'login': gen_string("alpha", 10),
+            'firstname': gen_string("alpha", 10),
+            'lastname': gen_string("alpha", 10),
+            'mail': '',
+            'password': gen_string("alpha", 10),
+            'auth-source-id': 1
+        }
+        result = UserObj().create(options)
+        self.assertNotEqual(result.return_code, 0)
+        self.assertTrue(result.stderr)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_user.py]: Is `assertTrue` the best assertion to use here? How about `self.assertGreaterThan(len(result.stderr), 0)` or `self.assertIn('desired error message', result.stdout)`.

File: tests/foreman/cli/test_user.py [hunk 6]
```diff
@@ -1657,6 +1681,42 @@ def test_list_user_4(self, test_data):
...
+        """
+        try:
+            user = make_user(test_data)
+        except CLIFactoryError as err:
+            self.fail(err)
+        self.__assert_exists(user)
+        result = UserObj.list({
+            u'search': u'mail = {0}'.format(test_data['mail']),
+        })
+        self.assertEqual(len(result.stderr), 0)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_user.py]: How about adding `self.assertEqual(result.status_code, 0)`?

File: tests/foreman/cli/test_user.py [hunk 7]
```diff
@@ -1657,6 +1681,42 @@ def test_list_user_4(self, test_data):
...
+        @Assert: User is listed
+
+        @BZ: 1204667
+
+        """
+        try:
+            user = make_user(test_data)
+        except CLIFactoryError as err:
+            self.fail(err)
+        self.__assert_exists(user)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_user.py]: Why is a method that begins with a double underscore used?

Comment (tkolhar) [inline tests/foreman/cli/test_user.py]: @Ichimonji10  there is already a method https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_user.py#L33 written i just used it

Comment (Ichimonji10) [inline tests/foreman/cli/test_user.py]: OK. That makes sense. Thank you.

**Comment by Ichimonji10** [comment]
ACK. Please rebase.

**Comment by elyezer** [comment]
ACK

**Pull request merged**